detektGenerateConfig doesn't work

## Expected Behavior
`./gradlew detektGenerateConfig` creates the default configuration when the configuration doesn't exist

## Observed Behavior
It doesn't, and it complains about it!

```
Execution failed for task ':app:detektGenerateConfig'.
> Provided path '/Users/brais.gabin/Workspace/project/config/detekt/detekt.yml' does not exist!
```

If I create the file with `touch /Users/brais.gabin/Workspace/project/config/detekt/detekt.yml` the task works but it doesn't creates the configuration file.

## Context

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:detektGenerateConfig'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:145)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:143)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
Caused by: org.gradle.api.GradleException: Provided path '/Users/brais.gabin/Workspace/project/config/detekt/detekt.yml' does not exist!
        at io.gitlab.arturbosch.detekt.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:65)
        at io.gitlab.arturbosch.detekt.invoke.DetektInvoker$DefaultImpls.invokeCli$default(DetektInvoker.kt:15)
        at io.gitlab.arturbosch.detekt.DetektGenerateConfigTask.generateConfig(DetektGenerateConfigTask.kt:65)
        at jdk.internal.reflect.GeneratedMethodAccessor1305.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
        at org.gradle.api.internal.tasks.execution.TaskExecution$2.run(TaskExecution.java:239)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:224)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:207)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:190)
        at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:168)
        at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:89)
        at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:40)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:53)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:50)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:50)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:40)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:48)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:36)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:29)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:61)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:42)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:60)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:27)
        at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:188)
        at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:75)
        at org.gradle.internal.Either$Right.fold(Either.java:175)
        at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:59)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:73)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:48)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:38)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:27)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:109)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:56)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:56)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:38)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:73)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:89)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:50)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:114)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:57)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:76)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:50)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:93)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:93)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:43)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:31)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
        at org.gradle.api.internal.tasks.execution.TaskExecution$3.withWorkspace(TaskExecution.java:284)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:33)
        at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:142)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
Caused by: java.lang.reflect.InvocationTargetException
        at jdk.internal.reflect.GeneratedMethodAccessor1306.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at io.gitlab.arturbosch.detekt.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:59)
        ... 119 more
Caused by: com.beust.jcommander.ParameterException: Provided path '/Users/brais.gabin/Workspace/project/config/detekt/detekt.yml' does not exist!
        at io.gitlab.arturbosch.detekt.cli.ExistingPathConverter.convert(ArgumentConverters.kt:17)
        at io.gitlab.arturbosch.detekt.cli.ExistingPathConverter.convert(ArgumentConverters.kt:12)
        at io.gitlab.arturbosch.detekt.cli.DetektInputPathConverter$convert$2.invoke(ArgumentConverters.kt:34)
        at io.gitlab.arturbosch.detekt.cli.DetektInputPathConverter$convert$2.invoke(ArgumentConverters.kt:34)
        at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:210)
        at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:786)
        at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:816)
        at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:807)
        at io.gitlab.arturbosch.detekt.cli.DetektInputPathConverter$DefaultImpls.convert(ArgumentConverters.kt:35)
        at io.gitlab.arturbosch.detekt.cli.MultipleExistingPathConverter.convert(ArgumentConverters.kt:44)
        at io.gitlab.arturbosch.detekt.cli.SpecKt$createSpec$1$5.invoke(Spec.kt:47)
        at io.gitlab.arturbosch.detekt.cli.SpecKt$createSpec$1$5.invoke(Spec.kt:42)
        at io.github.detekt.tooling.dsl.ProcessingSpecBuilder.config(ProcessingSpecBuilder.kt:32)
        at io.gitlab.arturbosch.detekt.cli.SpecKt$createSpec$1.invoke(Spec.kt:42)
        at io.gitlab.arturbosch.detekt.cli.SpecKt$createSpec$1.invoke(Spec.kt:9)
        at io.github.detekt.tooling.dsl.ProcessingSpecBuilderKt.processingSpec(ProcessingSpecBuilder.kt:15)
        at io.github.detekt.tooling.api.spec.ProcessingSpec$Companion.invoke(ProcessingSpec.kt:23)
        at io.gitlab.arturbosch.detekt.cli.SpecKt.createSpec(Spec.kt:9)
        at io.gitlab.arturbosch.detekt.cli.runners.ConfigExporter.execute(ConfigExporter.kt:16)
        ... 122 more
```

## Your Environment
* Version of detekt used: 1.20.0-RC1
* Version of Gradle used (if applicable): 7.3.1
* Operating System and version: MacOS
* Link to your project (if it's a public repository): private repo

--- a/detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/AnnotationExcluderSpec.kt
+++ b/detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/AnnotationExcluderSpec.kt
@@ -114,7 +114,7 @@ class AnnotationExcluderSpec(private val env: KotlinCoreEnvironment) {
         }
 
         @Nested
-        inner class `Know where ends a package` {
+        inner class `Know where a package ends` {
             val helloWorldAnnotationsKtFile = compileContentForTest(
                 """
                     package com.Hello

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliRunner.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliRunner.kt
@@ -20,7 +20,7 @@ class CliRunner : DetektCli {
 
         val specialRunner = when {
             arguments.showVersion -> VersionPrinter(outputChannel)
-            arguments.generateConfig -> ConfigExporter(arguments, outputChannel, errorChannel)
+            arguments.generateConfig -> ConfigExporter(arguments, outputChannel)
             arguments.printAst -> AstPrinter(arguments, outputChannel)
             else -> null
         }

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt
@@ -54,7 +54,7 @@ fun buildRunner(
     val arguments = parseArguments(args)
     return when {
         arguments.showVersion -> VersionPrinter(outputPrinter)
-        arguments.generateConfig -> ConfigExporter(arguments, outputPrinter, errorPrinter)
+        arguments.generateConfig -> ConfigExporter(arguments, outputPrinter)
         arguments.printAst -> AstPrinter(arguments, outputPrinter)
         else -> Runner(arguments, outputPrinter, errorPrinter)
     }

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporter.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporter.kt
@@ -14,7 +14,7 @@ class ConfigExporter(
     override fun execute() {
         val configPath = Paths.get(arguments.config ?: "detekt.yml")
         val spec = arguments.createSpec(outputPrinter, errorPrinter)
-        DefaultConfigurationProvider.load(spec).copy(configPath)
+        DefaultConfigurationProvider.load(spec.extensionsSpec).copy(configPath)
         outputPrinter.appendLine("Successfully copied default config to ${configPath.toAbsolutePath()}")
     }
 }

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporter.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporter.kt
@@ -1,19 +1,24 @@
 package io.gitlab.arturbosch.detekt.cli.runners
 
 import io.github.detekt.tooling.api.DefaultConfigurationProvider
+import io.github.detekt.tooling.api.spec.ProcessingSpec
 import io.gitlab.arturbosch.detekt.cli.CliArgs
-import io.gitlab.arturbosch.detekt.cli.createSpec
+import io.gitlab.arturbosch.detekt.cli.MultipleExistingPathConverter
 import java.nio.file.Paths
 
 class ConfigExporter(
     private val arguments: CliArgs,
     private val outputPrinter: Appendable,
-    private val errorPrinter: Appendable,
 ) : Executable {
 
     override fun execute() {
         val configPath = Paths.get(arguments.config ?: "detekt.yml")
-        val spec = arguments.createSpec(outputPrinter, errorPrinter)
+        val spec = ProcessingSpec {
+            extensions {
+                disableDefaultRuleSets = arguments.disableDefaultRuleSets
+                fromPaths { arguments.plugins?.let { MultipleExistingPathConverter().convert(it) }.orEmpty() }
+            }
+        }
         DefaultConfigurationProvider.load(spec.extensionsSpec).copy(configPath)
         outputPrinter.appendLine("Successfully copied default config to ${configPath.toAbsolutePath()}")
     }

--- a/detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporterSpec.kt
+++ b/detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporterSpec.kt
@@ -10,17 +10,13 @@ import java.nio.file.Files
 
 class ConfigExporterSpec {
 
-    @Nested
-    inner class `config exporter` {
+    @Test
+    fun `should export the given config`() {
+        val tmpConfig = createTempFileForTest("ConfigPrinterSpec", ".yml")
+        val cliArgs = parseArguments(arrayOf("--config", tmpConfig.toString()))
 
-        @Test
-        fun `should export the given config`() {
-            val tmpConfig = createTempFileForTest("ConfigPrinterSpec", ".yml")
-            val cliArgs = parseArguments(arrayOf("--config", tmpConfig.toString()))
+        ConfigExporter(cliArgs, NullPrintStream(), NullPrintStream()).execute()
 
-            ConfigExporter(cliArgs, NullPrintStream(), NullPrintStream()).execute()
-
-            assertThat(Files.readAllLines(tmpConfig)).isNotEmpty
-        }
+        assertThat(Files.readAllLines(tmpConfig)).isNotEmpty
     }
 }

--- a/detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporterSpec.kt
+++ b/detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/runners/ConfigExporterSpec.kt
@@ -4,7 +4,6 @@ import io.github.detekt.test.utils.NullPrintStream
 import io.github.detekt.test.utils.createTempFileForTest
 import io.gitlab.arturbosch.detekt.cli.parseArguments
 import org.assertj.core.api.Assertions.assertThat
-import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 import java.nio.file.Files
 
@@ -12,10 +11,10 @@ class ConfigExporterSpec {
 
     @Test
     fun `should export the given config`() {
-        val tmpConfig = createTempFileForTest("ConfigPrinterSpec", ".yml")
+        val tmpConfig = createTempFileForTest("ConfigPrinterSpec", ".yml").also { Files.delete(it) }
         val cliArgs = parseArguments(arrayOf("--config", tmpConfig.toString()))
 
-        ConfigExporter(cliArgs, NullPrintStream(), NullPrintStream()).execute()
+        ConfigExporter(cliArgs, NullPrintStream()).execute()
 
         assertThat(Files.readAllLines(tmpConfig)).isNotEmpty
     }

--- a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/ProcessingSettings.kt
+++ b/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/ProcessingSettings.kt
@@ -33,7 +33,7 @@ class ProcessingSettings(
     LoggingAware by LoggingFacade(spec.loggingSpec),
     PropertiesAware by PropertiesFacade(),
     EnvironmentAware by EnvironmentFacade(spec.projectSpec, spec.compilerSpec),
-    ClassloaderAware by ExtensionFacade(spec.extensionsSpec),
+    ClassloaderAware by ExtensionFacade(spec.extensionsSpec.plugins),
     SetupContext {
 
     override val configUris: Collection<URI> = spec.configSpec.extractUris()

--- a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/settings/ClassloaderAware.kt
+++ b/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/settings/ClassloaderAware.kt
@@ -14,11 +14,11 @@ interface ClassloaderAware {
 }
 
 class ExtensionFacade(
-    private val extensionsSpec: ExtensionsSpec
+    private val plugins: ExtensionsSpec.Plugins?
 ) : AutoCloseable, Closeable, ClassloaderAware {
 
     init {
-        extensionsSpec.plugins?.paths?.forEach {
+        plugins?.paths?.forEach {
             require(Files.exists(it)) { "Given plugin ‘$it’ does not exist." }
             require(it.toString().endsWith("jar")) { "Given plugin ‘$it’ is not a JAR." }
         }
@@ -28,7 +28,6 @@ class ExtensionFacade(
      * Shared class loader used to load services from plugin jars.
      */
     override val pluginLoader: ClassLoader by lazy {
-        val plugins = extensionsSpec.plugins
         when {
             plugins?.loader != null -> checkNotNull(plugins.loader)
             plugins?.paths != null -> {
@@ -46,7 +45,7 @@ class ExtensionFacade(
     }
 
     override fun closeLoaderIfNeeded() {
-        if (extensionsSpec.plugins?.paths != null) {
+        if (plugins?.paths != null) {
             // we created a classloader and need to close it
             closeQuietly(pluginLoader as? URLClassLoader)
         }

--- a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/DefaultConfigProvider.kt
+++ b/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/DefaultConfigProvider.kt
@@ -1,6 +1,7 @@
 package io.gitlab.arturbosch.detekt.core.tooling
 
 import io.github.detekt.tooling.api.DefaultConfigurationProvider
+import io.github.detekt.tooling.api.spec.ExtensionsSpec
 import io.github.detekt.tooling.api.spec.ProcessingSpec
 import io.github.detekt.utils.getSafeResourceAsStream
 import io.github.detekt.utils.openSafeStream
@@ -15,26 +16,26 @@ import java.nio.file.Path
 import java.nio.file.StandardCopyOption
 
 class DefaultConfigProvider : DefaultConfigurationProvider {
-    private lateinit var spec: ProcessingSpec
+    private lateinit var extensionsSpec: ExtensionsSpec
 
-    override fun init(spec: ProcessingSpec) {
-        this.spec = spec
+    override fun init(extensionsSpec: ExtensionsSpec) {
+        this.extensionsSpec = extensionsSpec
     }
 
-    override fun get(): Config = spec.getDefaultConfiguration()
+    override fun get(): Config = extensionsSpec.getDefaultConfiguration()
 
     override fun copy(targetLocation: Path) {
-        Files.copy(configInputStream(spec), targetLocation, StandardCopyOption.REPLACE_EXISTING)
+        Files.copy(configInputStream(extensionsSpec), targetLocation, StandardCopyOption.REPLACE_EXISTING)
     }
 }
 
-private fun configInputStream(spec: ProcessingSpec): InputStream {
+private fun configInputStream(extensionsSpec: ExtensionsSpec): InputStream {
     val outputStream = ByteArrayOutputStream()
 
-    requireNotNull(spec.javaClass.getSafeResourceAsStream("/default-detekt-config.yml"))
+    requireNotNull(extensionsSpec.javaClass.getSafeResourceAsStream("/default-detekt-config.yml"))
         .use { it.copyTo(outputStream) }
 
-    ExtensionFacade(spec.extensionsSpec.plugins).pluginLoader
+    ExtensionFacade(extensionsSpec.plugins).pluginLoader
         .getResourcesAsStream("config/config.yml")
         .forEach { inputStream ->
             outputStream.bufferedWriter().append('\n').flush()
@@ -50,6 +51,10 @@ private fun ClassLoader.getResourcesAsStream(name: String): Sequence<InputStream
         .map { it.openSafeStream() }
 }
 
-fun ProcessingSpec.getDefaultConfiguration(): Config {
+private fun ExtensionsSpec.getDefaultConfiguration(): Config {
     return YamlConfig.load(configInputStream(this).reader())
+}
+
+fun ProcessingSpec.getDefaultConfiguration(): Config {
+    return extensionsSpec.getDefaultConfiguration()
 }

--- a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/DefaultConfigProvider.kt
+++ b/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/DefaultConfigProvider.kt
@@ -34,7 +34,7 @@ private fun configInputStream(spec: ProcessingSpec): InputStream {
     requireNotNull(spec.javaClass.getSafeResourceAsStream("/default-detekt-config.yml"))
         .use { it.copyTo(outputStream) }
 
-    ExtensionFacade(spec.extensionsSpec).pluginLoader
+    ExtensionFacade(spec.extensionsSpec.plugins).pluginLoader
         .getResourcesAsStream("config/config.yml")
         .forEach { inputStream ->
             outputStream.bufferedWriter().append('\n').flush()

--- a/detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/tooling/DefaultConfigProviderSpec.kt
+++ b/detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/tooling/DefaultConfigProviderSpec.kt
@@ -11,11 +11,11 @@ import java.nio.file.Files
 class DefaultConfigProviderSpec {
     @Nested
     inner class `defaultConfigProvider without plugins` {
-        val spec = createNullLoggingSpec {}
+        private val extensionsSpec = createNullLoggingSpec {}.extensionsSpec
 
         @Test
         fun `gets`() {
-            val config = DefaultConfigProvider().apply { init(spec) }.get()
+            val config = DefaultConfigProvider().apply { init(extensionsSpec) }.get()
 
             assertThat(config.parentPath).isNull()
             assertThat(config.subConfig("build").valueOrNull<Int>("maxIssues")).isEqualTo(0)
@@ -25,7 +25,7 @@ class DefaultConfigProviderSpec {
         @Test
         fun `copies`() {
             val path = createTempFileForTest("test", "test")
-            DefaultConfigProvider().apply { init(spec) }.copy(path)
+            DefaultConfigProvider().apply { init(extensionsSpec) }.copy(path)
 
             assertThat(path)
                 .hasSameTextualContentAs(resourceAsPath("default-detekt-config.yml"))
@@ -34,15 +34,15 @@ class DefaultConfigProviderSpec {
 
     @Nested
     inner class `defaultConfigProvider with plugins` {
-        val spec = createNullLoggingSpec {
+        private val extensionsSpec = createNullLoggingSpec {
             extensions {
                 fromPaths { listOf(resourceAsPath("sample-rule-set.jar")) }
             }
-        }
+        }.extensionsSpec
 
         @Test
         fun `gets`() {
-            val config = DefaultConfigProvider().apply { init(spec) }.get()
+            val config = DefaultConfigProvider().apply { init(extensionsSpec) }.get()
 
             assertThat(config.parentPath).isNull()
             assertThat(config.subConfig("build").valueOrNull<Int>("maxIssues")).isEqualTo(0)
@@ -52,7 +52,7 @@ class DefaultConfigProviderSpec {
         @Test
         fun `copies`() {
             val path = createTempFileForTest("test", "test")
-            DefaultConfigProvider().apply { init(spec) }.copy(path)
+            DefaultConfigProvider().apply { init(extensionsSpec) }.copy(path)
 
             val actual = String(Files.readAllBytes(path), Charsets.UTF_8)
             val expected = String(Files.readAllBytes(resourceAsPath("default-detekt-config.yml")), Charsets.UTF_8) +

--- a/detekt-tooling/api/detekt-tooling.api
+++ b/detekt-tooling/api/detekt-tooling.api
@@ -11,12 +11,12 @@ public abstract interface class io/github/detekt/tooling/api/DefaultConfiguratio
 	public static final field Companion Lio/github/detekt/tooling/api/DefaultConfigurationProvider$Companion;
 	public abstract fun copy (Ljava/nio/file/Path;)V
 	public abstract fun get ()Lio/gitlab/arturbosch/detekt/api/Config;
-	public abstract fun init (Lio/github/detekt/tooling/api/spec/ProcessingSpec;)V
+	public abstract fun init (Lio/github/detekt/tooling/api/spec/ExtensionsSpec;)V
 }
 
 public final class io/github/detekt/tooling/api/DefaultConfigurationProvider$Companion {
-	public final fun load (Lio/github/detekt/tooling/api/spec/ProcessingSpec;Ljava/lang/ClassLoader;)Lio/github/detekt/tooling/api/DefaultConfigurationProvider;
-	public static synthetic fun load$default (Lio/github/detekt/tooling/api/DefaultConfigurationProvider$Companion;Lio/github/detekt/tooling/api/spec/ProcessingSpec;Ljava/lang/ClassLoader;ILjava/lang/Object;)Lio/github/detekt/tooling/api/DefaultConfigurationProvider;
+	public final fun load (Lio/github/detekt/tooling/api/spec/ExtensionsSpec;Ljava/lang/ClassLoader;)Lio/github/detekt/tooling/api/DefaultConfigurationProvider;
+	public static synthetic fun load$default (Lio/github/detekt/tooling/api/DefaultConfigurationProvider$Companion;Lio/github/detekt/tooling/api/spec/ExtensionsSpec;Ljava/lang/ClassLoader;ILjava/lang/Object;)Lio/github/detekt/tooling/api/DefaultConfigurationProvider;
 }
 
 public abstract interface class io/github/detekt/tooling/api/Detekt {

--- a/detekt-tooling/src/main/kotlin/io/github/detekt/tooling/api/DefaultConfigurationProvider.kt
+++ b/detekt-tooling/src/main/kotlin/io/github/detekt/tooling/api/DefaultConfigurationProvider.kt
@@ -1,13 +1,13 @@
 package io.github.detekt.tooling.api
 
-import io.github.detekt.tooling.api.spec.ProcessingSpec
+import io.github.detekt.tooling.api.spec.ExtensionsSpec
 import io.gitlab.arturbosch.detekt.api.Config
 import java.nio.file.Path
 import java.util.ServiceLoader
 
 interface DefaultConfigurationProvider {
 
-    fun init(spec: ProcessingSpec)
+    fun init(extensionsSpec: ExtensionsSpec)
 
     fun get(): Config
 
@@ -16,9 +16,12 @@ interface DefaultConfigurationProvider {
     companion object {
 
         fun load(
-            spec: ProcessingSpec,
+            extensionsSpec: ExtensionsSpec,
             classLoader: ClassLoader = DefaultConfigurationProvider::class.java.classLoader,
-        ): DefaultConfigurationProvider =
-            ServiceLoader.load(DefaultConfigurationProvider::class.java, classLoader).first().apply { init(spec) }
+        ): DefaultConfigurationProvider {
+            return ServiceLoader.load(DefaultConfigurationProvider::class.java, classLoader)
+                .first()
+                .apply { init(extensionsSpec) }
+        }
     }
 }

--- a/detekt-tooling/src/test/kotlin/io/github/detekt/tooling/api/DefaultConfigurationProviderSpec.kt
+++ b/detekt-tooling/src/test/kotlin/io/github/detekt/tooling/api/DefaultConfigurationProviderSpec.kt
@@ -1,16 +1,8 @@
 package io.github.detekt.tooling.api
 
 import io.github.detekt.test.utils.createTempFileForTest
-import io.github.detekt.tooling.api.spec.BaselineSpec
-import io.github.detekt.tooling.api.spec.CompilerSpec
-import io.github.detekt.tooling.api.spec.ConfigSpec
-import io.github.detekt.tooling.api.spec.ExecutionSpec
+import io.github.detekt.tooling.api.spec.ExtensionId
 import io.github.detekt.tooling.api.spec.ExtensionsSpec
-import io.github.detekt.tooling.api.spec.LoggingSpec
-import io.github.detekt.tooling.api.spec.ProcessingSpec
-import io.github.detekt.tooling.api.spec.ProjectSpec
-import io.github.detekt.tooling.api.spec.ReportsSpec
-import io.github.detekt.tooling.api.spec.RulesSpec
 import io.gitlab.arturbosch.detekt.api.Config
 import org.assertj.core.api.Assertions.assertThatCode
 import org.junit.jupiter.api.Nested
@@ -33,7 +25,7 @@ class DefaultConfigurationProviderSpec {
 }
 
 internal class TestConfigurationProvider : DefaultConfigurationProvider {
-    override fun init(spec: ProcessingSpec) {
+    override fun init(extensionsSpec: ExtensionsSpec) {
         // no-op
     }
 
@@ -44,23 +36,11 @@ internal class TestConfigurationProvider : DefaultConfigurationProvider {
     }
 }
 
-private object Spec : ProcessingSpec {
-    override val baselineSpec: BaselineSpec
+private object Spec : ExtensionsSpec {
+    override val disableDefaultRuleSets: Boolean
         get() = error("No expected call")
-    override val compilerSpec: CompilerSpec
-        get() = error("No expected call")
-    override val configSpec: ConfigSpec
-        get() = error("No expected call")
-    override val executionSpec: ExecutionSpec
-        get() = error("No expected call")
-    override val extensionsSpec: ExtensionsSpec
-        get() = error("No expected call")
-    override val rulesSpec: RulesSpec
-        get() = error("No expected call")
-    override val loggingSpec: LoggingSpec
-        get() = error("No expected call")
-    override val projectSpec: ProjectSpec
-        get() = error("No expected call")
-    override val reportsSpec: ReportsSpec
+    override val plugins: ExtensionsSpec.Plugins?
+        get() = null
+    override val disabledExtensions: Set<ExtensionId>
         get() = error("No expected call")
 }

